infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/TopologyDescriptor.java

```diff
@@ -1,91 +1,148 @@
 package org.corfudb.infrastructure.logreplication.infrastructure;
 
 import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
 import lombok.extern.slf4j.Slf4j;
 
 import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.TopologyConfigurationMsg;
 import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.ClusterRole;
 import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.ClusterConfigurationMsg;
 
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
- * This class represents a view of a Multi-Cluster/Site Topology,
- * where one cluster is the active/primary and n cluster's are standby's (backups).
+ * This class represents a view of a Multi-Cluster Topology,
+ *
+ * Ideally, in a given topology, one cluster represents the active cluster (source of data)
+ * while n others are standby clusters (backup's). However, because the topology info is provided by an
+ * external adapter which can be specific to the use cases of the user, a topology might be initialized
+ * with multiple active clusters and multiple standby clusters.
+ *
  */
 @Slf4j
 public class TopologyDescriptor {
 
     // Represents a state of the topology configuration (a topology epoch)
     @Getter
+    @Setter
```

Comment:
Is this setter necessary? I think we should use a new object when we need to update config id...

Reply:
+1, changed.